apps/wolfssh/common.c

```diff
@@ -186,20 +187,179 @@ void ClientIPOverride(int flag)
 #endif /* WOLFSSH_CERTS */
 
 
+static int AppendKeyToFile(const char* filename, const char* name,
+        const char* type, const char* key)
+{
+    FILE *f;
+
+    f = fopen(filename, "a");
+    fprintf(f, "%s %s %s\n", name, type, key);
+    fclose(f);
+
+    return 0;
+}
+
+
+static int FingerprintKey(const byte* pubKey, word32 pubKeySz, char* out)
+{
+    wc_Sha256 sha;
+    byte digest[WC_SHA256_DIGEST_SIZE];
+    char fp[48] = { 0 };
+    word32 fpSz = sizeof(fp);
+
+    wc_InitSha256(&sha);
+    wc_Sha256Update(&sha, pubKey, pubKeySz);
+    wc_Sha256Final(&sha, digest);
+
+    Base64_Encode_NoNl(digest, sizeof(digest), (byte*)fp, &fpSz);
+
+    if (fp[fpSz] == '=') {
+        fp[fpSz] = 0;
+    }
+
+    strcat(out, "SHA256:");
+    strcat(out, fp);
+
+    return 0;
+}
+
+
+static int GetConfirmation(void)
+{
+    int c, confirmed = 0;
+
+    printf("Y/n: ");
+    c = getchar();
+
+    if (c == 'Y') {
+        confirmed = 1;
+    }
+
+    return confirmed;
+}
+
+
 int ClientPublicKeyCheck(const byte* pubKey, word32 pubKeySz, void* ctx)
 {
-    int ret = 0;
+    char *cursor;
+    char *line;
+    const char *targetName = (const char*)ctx;
+    char *name;
+    char *keyType;
+    char *key;
+    char *knownHosts = NULL;
+    char *knownHostsName = NULL;
+    int ret = 0, found = 0, badMatch = 0, otherMatch = 0;
+    word32 sz, lineCount = 0;
+    char encodedKey[1200] = { 0 };
```

Comment:
pretty large stack value and magic number, needs a small stack option and macro or comment on value choosen

Reply:
The stack wasn't a concern anywhere else in the application. I think this is a case where it should be allocated, though.